src/input-select/input-range-select.tsx

```diff
@@ -0,0 +1,316 @@
+import { InputSelectWrapper } from "./input-select-wrapper";
+import {
+    Divider,
+    LabelContainer,
+    PlaceholderLabel,
+    RangeIcon,
+    Selector,
+    ValueLabel,
+} from "./input-select.styles";
+import { InputRangeProp, InputRangeSelectProps } from "./types";
+
+type RangeType = "from" | "to";
+
+export const InputRangeSelect = <T, V>({
+    selectedOptions,
+    placeholders = { from: "Select", to: "Select" },
+    options,
+    disabled,
+    readOnly,
+    error,
+    "data-testid": testId,
+    id,
+    enableSearch = false,
+    searchFunction,
+    searchPlaceholder,
+    valueExtractor,
+    valueToStringFunction,
+    listExtractor,
+    displayValueExtractor,
+    onSelectOption,
+    listStyleWidth,
+    onShowOptions,
+    onHideOptions,
+    onRetry,
+    optionsLoadState = { from: "success", to: "success" },
+    optionTruncationType = "middle",
+    renderCustomSelectedOption,
+    renderListItem,
+    renderCustomCallToAction,
+    ...otherProps
+}: InputRangeSelectProps<T, V>): JSX.Element => {
+    // =============================================================================
+    // CONST, STATE
+    // =============================================================================
+    const [selected, setSelected] = useState<InputRangeProp<T>>({
+        from: selectedOptions?.from,
+        to: selectedOptions?.to,
+    });
+    const [showOptions, setShowOptions] = useState<InputRangeProp<boolean>>({
+        from: false,
+        to: false,
+    });
+
+    const selectorRef = useRef<HTMLButtonElement>();
+    const labelContainerRef = {
+        from: useRef<HTMLDivElement>(),
+        to: useRef<HTMLDivElement>(),
+    };
+    const [currentElementType, setCurrentElementType] = useState<
+        "start" | "end" | "none"
+    >("none");
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        setSelected({ from: selectedOptions?.from, to: selectedOptions?.to });
+    }, [selectedOptions]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleSelectorClick = (
+        event: React.MouseEvent,
+        rangeType?: RangeType
+    ) => {
+        event.stopPropagation();
+        event.preventDefault();
+        if (disabled || readOnly) {
+            return;
+        }
+        if (rangeType === "from") {
+            showDropDown("from");
+        } else if (rangeType === "to" && selected.from) {
+            showDropDown("to");
+        } else if (rangeType === "to" && !selected.from) {
+            showDropDown("from");
+        } else {
+            showDropDown("from");
+        }
+    };
+
+    const showDropDown = (rangeType: RangeType) => {
```

Comment:
with the merging of the state, you do not need this function anymore

Reply:
Removed